### ISSUE TITLE
Multi language blog

### ISSUE THREAD

**Issue by marlonpatrick10** [LouisBarranqueiro/tranquilpeak-hexo-theme#48]
Hi Louis, I doing some modifications in your theme for two purposes:

- Internationalize the theme
- Make a multi language blog

I open this issue for describe what I changed and for get your feedback if my method is the best form:

**Internationalize**

- Make a folder language in theme folder: tranquilpeak/languages
- Inside language folder a put two files, one for each language: pt-br.yml and en.yml. The en.yml file is something like this:
```yaml
post_meta_date_format: MMM DD, YYYY
all_categories_date_format: MMM DD, YYYY
all_tags_date_format: MMM DD, YYYY
all_archives_date_format: MMM DD, YYYY
all_archives_month_format: MMMM

label_newer_posts: NEWER POSTS
label_older_posts: OLDER POSTS
label_page: Page
label_of: of
label_in: in
label_search_category: Search Category
label_search_tag: Search Tag
label_search_date: "Search Date (YEARMONTHDAY)"

read_more_message: Continue reading
go_to_message: Go to the website

author:
    email: myemail@email.com
    bio: My bio
    job: My job
    location: My Location

sidebar:
    menu:
        home:
            title: Home
            url: /en
        categories:
            title: Categories
            url: /all-categories-en
```

- Change fix date formats in all files. For example, in the file tranquilpeak/layout/_partial/post/meta.ejs:
```javascript

post.date.format('MMM DD, YYYY') //Before

post.date.locale(post.lang).format(__('post_meta_date_format')) //After
```

- Change fix strings in all files. For example, in the file tranquilpeak/layout/_partial/post/actions.ejs:
```html

<span class="hide-xs hide-sm text-small icon-ml">PREVIOUS</span>  <!-- Before -->

<span class="hide-xs hide-sm text-small icon-ml"><%= __('label_previous') %></span> <!-- After -->
```
- Change the _config.yml for setting the sidebar menu and others options. For example:

```yaml
sidebar:
    menu:
        home:
            title: sidebar.menu.home.title # this is a key for a language file
            url: sidebar.menu.home.url # this is a key for a language file
            icon: home
```

** Multi language blog**

- For each post or page, setting the language. For example, all-archives-pt-br/index.md:

```yaml
title: "all-archives"
layout: "all-archives"
lang: pt-br
---
```
- In blog _confg.yml setting the languages and permalinks. For example:

```yaml
language:
 - pt-br
 - en

permalink: :lang/:year/:month/:day/:title/
```

- Change templates for show only the current language posts. For example, in the file tranquilpeak/layout/_partial/index.ejs:

```html
<section class="post-group main-content-wrap">
    <% page.posts.each(function(post) { %>
	<% if (page.lang == post.lang) { %>
``` 

- Register a index generator for the second language, in my case english. File node_modules/hexo-generator-index/index.ejs:

```javascript
hexo.extend.generator.register('index-en', require('./lib/generator-en'));
```
- Create a index generator for the second language, in my case english. File node_modules/hexo-generator-index/lib/generator-en.js:

```javascript
'use strict';

var pagination = require('hexo-pagination');

module.exports = function(locals){
  var config = this.config;
  var posts = locals.posts.find({lang: 'en'}).sort('-date');
  var paginationDir = config.pagination_dir || 'page';

  return pagination('en', posts, {
    perPage: config.index_generator.per_page,
    layout: ['index', 'archive'],
    format: paginationDir + '/%d/',
    data: {
      __index: true
    }
  });
};
```
- Create a index page for the second language, in my case english, where the name is in agree with generator-en.ejs `return pagination('en', posts, {>`. In this case the page name shold be en and the layout should be index: `hexo new page en --lang en`.
```yaml
layout: index
lang: en
---
```

In an general view this was the process for create multi language blog. Can you help validate if this methodology is the best way? Can you think others ways more easier?

**Comment by LouisBarranqueiro** [comment]
You did such a good job. I appreciate a lot your involvement! :) I thing, we have to go step by step to implement these features beacause we have to greatly edit the structure. Hexo supports [Localization (i18n)](https://hexo.io/docs/localization.html), so the first is step is to implement this one by respecting the structure to have a blog internationalized (except articles). Will you be able to do that and pull request? :)

About internationalization of articles, I think it will be better and useful for the community to propose directly an implementation in Hexo. If we implement this feature in tranquipeak theme, I think that its compatibility with new features or behaviors of new Hexo versions will be reduced or maybe even become incompatible. Your idea is great! And I think it has to be implemented in Hexo in my opinion.

**Comment by marlonpatrick10** [comment]
Hi Louis, the last days I'm very occupied but I will try send pull request until the next week with the internationalization.

**Comment by LouisBarranqueiro** [comment]
Hi patrick, calm down man :D take your time :)

**Comment by LouisBarranqueiro** [comment]
This feature is now available in [Tranquilpeak v1.3.0](https://github.com/LouisBarranqueiro/tranquilpeak-hexo-theme/releases/tag/v1.3.0)
